### PR TITLE
WIP: Add how it works component 

### PR DESCRIPTION
**Pull request by ranwahle**
**What kind of change does this PR introduce?** (check _one_ with "x")
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Docs related changes
[ ] Other... Please describe:
```

**Link to the open issue**:
#158

### CODE REVIEW

**Comment by nirgn975** [comment]
Your master is not update, go to master `pull` and then merge it to your branch.
There is a lot of code here that shouldn't be changed.

**Comment by ranwahle** [comment]
I did it before commiting

**Comment by nirgn975** [comment]
Look at the [Files changed](https://github.com/hasadna/open_pension/pull/239/files) in your PR, and the actual files (for example [angular-cli.json](https://github.com/hasadna/open_pension/blob/master/client/angular-cli.json), there're changes that shouldn't be there..

Check the files again.

File: client/angular-cli.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "project": {
-    "version": "1.0.0-beta.26",
```
Comment (nirgn975) [inline client/angular-cli.json]: Way you changed this?

File: client/angular-cli.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-    "name": "openpension"
```
Comment (nirgn975) [inline client/angular-cli.json]: Way you changed this?

File: client/angular-cli.json [hunk 3]
```diff
@@ -18,16 +18,27 @@
       "prefix": "op",
       "mobile": false,
       "styles": [
+        "../node_modules/bootstrap/dist/css/bootstrap.css",
```
Comment (nirgn975) [inline client/angular-cli.json]: Bootstrap is a dependency of intro?

File: client/angular-cli.json [hunk 4]
```diff
@@ -18,16 +18,27 @@
...
+        "../node_modules/intro.js/introjs.css",
+        "../node_modules/intro.js/introjs-rtl.css",
         "styles.scss"
       ],
-      "scripts": [],
+      "scripts": [
+        "../node_modules/jquery/dist/jquery.js",
```
Comment (nirgn975) [inline client/angular-cli.json]: jquery is a dependency of intro?

File: client/angular-cli.json [hunk 5]
```diff
@@ -18,16 +18,27 @@
...
+        "../node_modules/tether/dist/js/tether.js",
+        "../node_modules/bootstrap/dist/js/bootstrap.js",
```
Comment (nirgn975) [inline client/angular-cli.json]: bootstrap is a dependency of intro?

File: client/angular-cli.json [hunk 6]
```diff
@@ -18,16 +18,27 @@
...
+        "../node_modules/intro.js/intro.js"
+
```
Comment (nirgn975) [inline client/angular-cli.json]: Why the whitespace?

File: client/angular-cli.json [hunk 7]
```diff
@@ -18,16 +18,27 @@
...
+      ],
       "environments": {
         "source": "environments/environment.ts",
         "dev": "environments/environment.ts",
         "prod": "environments/environment.prod.ts"
       }
     }
   ],
+  "addons": [],
```
Comment (nirgn975) [inline client/angular-cli.json]: Why you added this?

File: client/angular-cli.json [hunk 8]
```diff
@@ -18,16 +18,27 @@
...
+  "packages": [],
```
Comment (nirgn975) [inline client/angular-cli.json]: Why you added this?

File: client/package.json [hunk 9]
```diff
@@ -1,68 +1,52 @@
 {
-  "name": "openpension",
```
Comment (nirgn975) [inline client/package.json]: Why you change it?

File: client/package.json [hunk 10]
```diff
@@ -1,68 +1,52 @@
...
+  "name": "op",
   "version": "1.0.0",
-  "license": "ISC",
```
Comment (nirgn975) [inline client/package.json]: Why you change it?

File: client/package.json [hunk 11]
```diff
@@ -1,68 +1,52 @@
...
+  "license": "MIT",
   "angular-cli": {},
   "scripts": {
-    "ng": "ng",
     "start": "ng serve",
-    "lint": "tslint \"src/**/*.ts\" --project src/tsconfig.json --type-check && tslint \"e2e/**/*.ts\" --project e2e/tsconfig.json --type-check",
```
Comment (nirgn975) [inline client/package.json]: Why you change it?

File: client/package.json [hunk 12]
```diff
@@ -1,68 +1,52 @@
 {
-  "name": "openpension",
+  "name": "op",
   "version": "1.0.0",
-  "license": "ISC",
+  "license": "MIT",
   "angular-cli": {},
   "scripts": {
-    "ng": "ng",
```
Comment (nirgn975) [inline client/package.json]: Why you remove it?

File: client/package.json [hunk 13]
```diff
@@ -1,68 +1,52 @@
...
     "start": "ng serve",
-    "lint": "tslint \"src/**/*.ts\" --project src/tsconfig.json --type-check && tslint \"e2e/**/*.ts\" --project e2e/tsconfig.json --type-check",
+    "lint": "tslint \"src/**/*.ts\"",
     "test": "ng test",
-    "pree2e": "webdriver-manager update --standalone false --gecko false",
```
Comment (nirgn975) [inline client/package.json]: Why you change it?

File: client/package.json [hunk 14]
```diff
@@ -1,68 +1,52 @@
...
+    "pree2e": "webdriver-manager update",
     "e2e": "protractor"
   },
-  "private": false,
```
Comment (nirgn975) [inline client/package.json]: Why you change it?

File: client/package.json [hunk 15]
```diff
@@ -1,68 +1,52 @@
...
+  "private": true,
   "dependencies": {
-    "@angular/common": "^2.3.1",
```
Comment (nirgn975) [inline client/package.json]: We're on angular 2.3, not 2.1

File: client/package.json [hunk 16]
```diff
@@ -1,68 +1,52 @@
...
+    "@angular/forms": "~2.1.0",
+    "@angular/http": "~2.1.0",
+    "@angular/platform-browser": "~2.1.0",
+    "@angular/platform-browser-dynamic": "~2.1.0",
+    "@angular/router": "~3.1.0",
     "@ngrx/core": "^1.2.0",
     "@ngrx/effects": "^2.0.0",
     "@ngrx/store": "^2.2.1",
     "@ngrx/store-devtools": "^3.2.2",
-    "@types/d3-hierarchy": "^1.0.4",
```
Comment (nirgn975) [inline client/package.json]: Why you remove the `@types`?

File: client/angular-cli.json (hunk 3 above)
Comment (ranwahle) [inline client/angular-cli.json]: No. Removed, together with jQuery and tether

File: client/package.json [hunk 17]
```diff
@@ -33,6 +33,7 @@
     "@types/d3-selection": "^1.0.6",
     "@types/d3-shape": "^1.0.6",
     "@types/d3-transition": "^1.0.6",
+    "bootstrap": "^4.0.0-alpha.5",
```
Comment (nirgn975) [inline client/package.json]: Why there is a need for bootstrap?

File: client/package.json [hunk 18]
```diff
@@ -66,3 +68,4 @@
     "typescript": "~2.0.3"
   }
 }
+
```
Comment (nirgn975) [inline client/package.json]: Why the white space?

File: client/src/app/app-routing.module.ts [hunk 19]
```diff
@@ -1,17 +1,13 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
-import { PaiComponent } from './components/pai/pai.component';
-import { AboutComponent } from './components/about/about.component';
-import { PostsComponent } from './components/posts/posts.component';
-import { PostComponent } from './components/post/post.component';
+import { PaiComponent, HowItWorksComponent } from './components';
+
```
Comment (nirgn975) [inline client/src/app/app-routing.module.ts]: Why the whitespace?

File: client/src/app/app.module.ts [hunk 20]
```diff
@@ -16,14 +16,14 @@ import { AppRoutingModule } from './app-routing.module';
 import { OpComponent } from './op.component';
 import reducer from './reducers';
 
-import { HeaderComponent } from './components/header/header.component';
```
Comment (nirgn975) [inline client/src/app/app.module.ts]: Why change the imports? There shouldn't be a `inedx.ts` file in `components` directory so your `import` shouldn't work.

File: client/src/app/app.module.ts [hunk 21]
```diff
@@ -39,9 +39,7 @@ import { PostsComponent } from './components/posts/posts.component';
     FooterComponent,
     PaiComponent,
     FiltersComponent,
-    AboutComponent,
```
Comment (nirgn975) [inline client/src/app/app.module.ts]: Why you removed `AboutComponent`, `PostComponent`, and `PostsComponent` components?

File: client/src/app/components/how-it-works/how-it-works.component.scss [hunk 22]
```diff
@@ -0,0 +1,23 @@
+$grayColor: #B0BFC7;
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.scss]: Please define it in the `vars.scss` file (in `assets` directory) and `import` it here if you need it.

File: client/src/app/components/how-it-works/how-it-works.component.scss [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+.parameterBulit {
+  background-color: $grayColor;
+  color: white;
+  border-color: gray;
+  border-radius: 5px;
+  padding: 5px;
+  margin-bottom: 10px;
+
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.scss]: Remove the whitespace

File: client/src/app/components/how-it-works/how-it-works.component.scss [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+}
+
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.scss]: Remove the whitespace

File: client/src/app/components/how-it-works/how-it-works.component.ts [hunk 25]
```diff
@@ -0,0 +1,28 @@
+import {Component, OnInit, AfterViewInit} from '@angular/core';
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.ts]: Please add whitespace after `{` and before the closing `}`

File: client/src/app/components/how-it-works/how-it-works.component.ts [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+
+declare var introJs;
+
+@Component({
+  selector: 'op-how-it-works',
+  templateUrl: './how-it-works.component.html',
+  styleUrls: ['./how-it-works.component.scss']
+})
+export class HowItWorksComponent implements OnInit, AfterViewInit {
+  ngAfterViewInit(): void {
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.ts]: What's the reason for using `ngAfterViewInit` and not `ngOnInit`?

File: client/src/app/components/how-it-works/how-it-works.component.ts [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  intros: string[] = [
+    'כדי לחזור אחורה בעומק הפאי, אפשר ללחוץ על הפרמטרים העליונים שכבר נעשתה בהם בחירה',
+    'כדי לראות את הפרמטר הבא ברשימה צריך ללחוץ על פלח במעגל הפנימי ואז הוא יהפוך להיות מרכז המעגל ופרמטר נוסף יתווסף כמעגל חיצוני מרשימת הפרמטרים '
+
+  ]
+  constructor() { }
+
+  ngOnInit() {
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.ts]: If you don't use it, remove it.

File: client/src/app/components/index.ts [hunk 28]
```diff
@@ -0,0 +1,6 @@
+export * from './header/header.component';
```
Comment (nirgn975) [inline client/src/app/components/index.ts]: This file shouldn't be exist.

File: client/src/app/components/how-it-works/how-it-works.component.html [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+      </div>
+    </div>
+
+  </div>
+</div>
+
+<div class="paiPart"
+     data-intro="כדי לראות את הפרמטר הבא ברשימה צריך ללחוץ על פלח במעגל הפנימי ואז הוא יהפוך להיות מרכז המעגל ופרמטר נוסף יתווסף כמעגל חיצוני מרשימת הפרמטרים">
+  <op-pai></op-pai>
+
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.html]: Why the whitespace?

File: client/src/app/components/how-it-works/how-it-works.component.html [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+         data-intro="כדי לחזור אחורה בעומק הפאי, אפשר ללחוץ על הפרמטרים העליונים שכבר נעשתה בהם בחירה">
+      בסיס
+      <br/>
+      <span style="font-size: 0.5em;">
+         סה"כ שוק הפנסיה 100%
+    </span>
+    </div>
+    <div data-intro="זו רשימת הפרמטרים שיופיעו בגרף
+.כל פעם יופיעו בגרף 4 פרמטרים (אלה הצבעוניים בהתאמה)
+
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.html]: Why this is stretched over couple of lines?

File: client/src/app/components/how-it-works/how-it-works.component.html [hunk 31]
```diff
@@ -0,0 +1,44 @@
+<div style="direction: rtl;">
```
Comment (nirgn975) [inline client/src/app/components/how-it-works/how-it-works.component.html]: Remove the `direction`, there should be a direction for the whole website (you can see it [here](https://github.com/hasadna/open_pension/blob/master/client/src/styles.scss#L6))

File: client/package.json (hunk 18 above)
Comment (ranwahle) [inline client/package.json]: Gone

File: client/angular-cli.json [hunk 32]
```diff
@@ -18,9 +18,12 @@
       "prefix": "op",
       "mobile": false,
       "styles": [
+        "../node_modules/intro.js/introjs.css",
```
Comment (nirgn975) [inline client/angular-cli.json]: Did you try to check for a introjs typescript version?
Try `npm install --save-dev @types/intro.js` (or `npm install --save-dev @types/introjs`)

**Pull request closed without merging**